src/docfx/docset/Document.cs

```diff
@@ -194,7 +205,7 @@ public override string ToString()
         /// </summary>
         /// <param name="docset">The current docset</param>
         /// <param name="path">The path relative to docset root</param>
-        public static (Error error, Document doc) TryCreate(Docset docset, string path, string redirectionUrl = null)
+        public static (Error error, Document doc) TryCreate(Docset docset, string path, string redirectionUrl = null, bool isFromHistory = false)
```

Comment:
should this method be 
```csharp
public static bool TryCreate(Docset docset, string path, out Error error, out Document doc, string redirectionUrl = null, bool isFromHistory = false)
```
Then, we do not need to check error again at outer layer.